torch_geometric/data/summary.py

```diff
@@ -89,4 +115,24 @@ def __repr__(self) -> str:
             content.append(row)
         body = tabulate(content, headers='firstrow', tablefmt='psql')
 
+        if len(self.num_nodes_per_type) > 0 and \
+                len(self.num_edges_per_type) > 0:
```

Comment:
I believe it's always recommended to check whether list is empty using just `if list_name` rather than comparing its length with zero
```suggestion
        if self.num_nodes_per_type and \
                self.num_edges_per_type:
```